src/lib/logger.ts

```diff
@@ -0,0 +1,388 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import fs from 'fs';
+import os, { userInfo } from 'os';
+import path from 'path';
+import { Writable } from 'stream';
+import util from 'util';
+import winston from 'winston';
+import Transport from 'winston-transport';
+import env from './env';
+
+interface VIPLoggerOptions {
+	logDir?: string;
+	consoleLevel?: string;
+	fileLevel?: string;
+}
+
+interface DebugLikeLogger {
+	( ...args: unknown[] ): boolean;
+	enabled: boolean;
+	namespace: string;
+	error: ( ...args: unknown[] ) => boolean;
+	warn: ( ...args: unknown[] ) => boolean;
+	info: ( ...args: unknown[] ) => boolean;
+	verbose: ( ...args: unknown[] ) => boolean;
+	debug: ( ...args: unknown[] ) => boolean;
+	silly: ( ...args: unknown[] ) => boolean;
+}
+
+class VIPLogger {
+	private readonly logDir: string;
+	private readonly rootLogger: winston.Logger;
+	private readonly loggers: Map< string, DebugLikeLogger >;
+	private readonly enabledNamespaces: Set< string >;
+	private readonly isDebugMode: boolean;
+	private readonly commandName: string;
+	private readonly logFilePath: string;
+
+	constructor( options: VIPLoggerOptions = {} ) {
+		this.logDir = options.logDir || path.join( os.homedir(), '.vip-cli', 'logs' );
+		fs.mkdirSync( this.logDir, { recursive: true } );
+
+		// Check if --debug flag is passed
+		this.isDebugMode = process.argv.includes( '--debug' ) || process.argv.includes( '-d' );
+
+		// Container for namespace-specific loggers
+		this.loggers = new Map();
+
+		// Parse DEBUG env var like the debug library does
+
+		this.enabledNamespaces = new Set( [ '*' ] ); // Enable all by default
+		if ( process.env.DEBUG ) {
+			this.enabledNamespaces = this._parseDebugEnv( process.env.DEBUG );
+		}
+
+		// Determine command name from process.argv
+		this.commandName = this._determineCommandName();
+
+		// Create log file path with timestamp and command name
+		this.logFilePath = this._createLogFilePath();
+
+		// Create the root logger with shared transports
+		this.rootLogger = this._createRootLogger( options );
+	}
+
+	private _determineCommandName(): string {
```

Comment:
This is a garbage function and needs a bit improvement before it's ready for prod.